Upgrade to PHP 8.5 with multi-architecture support and configuration improvements

azinchen opened the pull request:
## Major Changes
- Upgrade from PHP 7 to PHP 8.5
- Update Alpine base image to 3.23.4
- Migrate all PHP configurations from php7 to php85 directories
- Update TorrentMonitor source to use php85-compatibility branch

## Multi-Architecture Support
- Add support for linux/s390x and linux/riscv64 platforms
- Build gnu-libiconv 1.15 from source with musl libc compatibility
- Manually patch lib/loop_wchar.h to fix mbrtowc() declaration conflict
- Configure build to use amd64 only on non-master branches for testing

## Configuration Enhancements
- Add NGINX_PORT environment variable for configurable port (default: 80)
- Document NGINX_PORT usage in README files
- Add complete platform support list to documentation

## Version Tracking Improvements
- Separate TorrentMonitor app version from Dockerized container version
- Renamed VERSION/RELEASE_DATE to TM_VERSION/TM_RELEASE_DATE (for upstream PHP app)
- Added DOCKERIZED_VERSION and BUILD_DATE build arguments (for container version)
- Updated deploy.yml to pass build args with runtime-generated ISO 8601 timestamps
- Updated torrentmonitor_update.yml to use new variable names
- Added custom OCI labels: torrentmonitor.app.version and torrentmonitor.app.release-date
- OCI image labels now correctly reference Dockerized version and build date

## Docker Improvements
- Update Dockerfile labels to OCI standard schema
- Replace deprecated 'maintainer' label with 'org.opencontainers.image.authors'
- Add OCI metadata labels (version, created, url, title, description)
- Remove obsolete docker-compose.yml version field

## Code Quality Improvements
- Fix nginx.conf formatting - standardize spacing between directives and values
- Fix default.conf formatting - standardize indentation to 4 spaces and remove extra spacing
- Fix typo in PHP configuration: max_executionn_time -> max_execution_time

## Files Changed
- .github/workflows/deploy.yml: Multi-arch build configuration, build arguments
- .github/workflows/torrentmonitor_update.yml: Updated version variable names
- Dockerfile: PHP 8.5 packages, gnu-libiconv build, OCI labels, version tracking
- README-RU.md & README.md: NGINX_PORT documentation and platform list
- rootfs/etc/nginx/nginx.conf: Fixed formatting consistency
- rootfs/etc/nginx/conf.d/default.conf: Fixed indentation and spacing
- rootfs/etc/php85/*: PHP 8.5 configuration files (migrated from php7, fixed typos)
- rootfs/init: Updated to use php85-fpm binary and paths

Fixes azinchen/torrentmonitor-dockerized#112

azinchen committed:
Upgrade to PHP 8.5 with multi-architecture support and configuration improvements

Major Changes:
- Upgrade from PHP 7 to PHP 8.5
- Update Alpine base image to 3.23.2
- Migrate all PHP configurations from php7 to php85 directories
- Update TorrentMonitor source to use php85-compatibility branch

Multi-Architecture Support:
- Add support for linux/amd64, linux/arm64, and linux/arm/v7 platforms
- Build gnu-libiconv 1.15 from source with musl libc compatibility
- Manually patch lib/loop_wchar.h to fix mbrtowc() declaration conflict
- Configure build to use amd64 only on non-master branches for testing

Configuration Enhancements:
- Add NGINX_PORT environment variable for configurable port (default: 80)
- Update docker-compose.yml to use NGINX_PORT variable
- Document NGINX_PORT usage in README files
- Add complete platform support list to documentation

Docker Improvements:
- Update Dockerfile labels to OCI standard schema
- Replace deprecated 'maintainer' label with 'org.opencontainers.image.authors'
- Add OCI metadata labels (version, created, url, title, description)
- Remove obsolete docker-compose.yml version field
- Remove unused patch files in favor of direct sed patching

Files Changed:
- .github/workflows/deploy.yml: Multi-arch build configuration
- Dockerfile: PHP 8.5 packages, gnu-libiconv build, OCI labels
- README-RU.md & README.md: NGINX_PORT documentation and platform list
- rootfs/etc/php85/*: PHP 8.5 configuration files (migrated from php7)
- rootfs/init: Updated to use php85-fpm binary and paths

Simplify libiconv patch: remove line directly instead of conditional compilation

Since all target platforms are Linux-based, the conditional compilation
directives are unnecessary. Simply delete line 40 from loop_wchar.h.

Simplify sed command by specifying full path

Separate TorrentMonitor app and Dockerized versions

- Renamed VERSION/RELEASE_DATE to TM_VERSION/TM_RELEASE_DATE for TorrentMonitor app
- Added DOCKERIZED_VERSION and BUILD_DATE build args for container versioning
- Updated deploy.yml to pass build args with runtime-generated timestamps
- Updated torrentmonitor_update.yml to use new variable names
- Added custom labels: torrentmonitor.app.version and torrentmonitor.app.release-date
- OCI labels now correctly reference Dockerized version and build date

Fix nginx.conf formatting - standardize spacing

Fix default.conf formatting - standardize indentation and spacing

Fix gnu-libiconv build for Alpine 3.23.2 compatibility

- Update sed command to remove lines 39-48 from loop_wchar.h
- Removes conflicting mbrtowc/mbsinit declarations that clash with musl libc
- Resolves compilation errors with modern Alpine 3.23.2 headers
- Successfully builds gnu-libiconv-1.15-r3 package

Fix BUILD_DATE by setting it as environment variable in workflow

Add TorrentMonitor app URL and source labels to Dockerfile

Fix formatting: remove trailing slash from URL and extra spaces

Update Dockerfile

azinchen committed:
Update Alpine base image from 3.23.2 to 3.23.3

azinchen committed:
chore: bump Alpine base image to 3.23.4

azinchen commented:
final rebase before closing the pr

alfonder commented on rootfs/etc/nginx/conf.d/default.conf:
```diff
@@ -1,42 +1,42 @@
+        log_not_found off;
     }
-  }
+}
```
This change contains no behavioral changes, but only formatting/indentation. Please, revert this file back to keep the history cleaner and more readable.

alfonder commented on rootfs/etc/nginx/nginx.conf:
```diff
@@ -18,15 +18,15 @@ http {
 
-    access_log  off;
+    access_log off;
     error_log /proc/1/fd/2 error;
 
-    sendfile        on;
-    #tcp_nopush     on;
+    sendfile on;
+    #tcp_nopush on;
 
-    keepalive_timeout  65;
+    keepalive_timeout 65;
 
-    #gzip  on;
+    #gzip on;
 
     include /etc/nginx/conf.d/*.conf;
 }
```
This change contains no behavioral changes, but only formatting/indentation. Please, revert this file back to keep the history cleaner and more readable.

alfonder commented on .github/workflows/deploy.yml:
```diff
@@ -32,10 +32,19 @@ jobs:
           echo "image_tag=${IMAGE_TAG}" >> $GITHUB_ENV
           echo "is_tag=${IS_TAG}" >> $GITHUB_ENV
+          echo "build_platforms=${BUILD_PLATFORMS}" >> $GITHUB_ENV
+          echo "build_date=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
```
I agree with Copilot. The BUILD_DATE variable is used in line 47. Therefore I suggest to accept the recommended change.

alfonder replied:
Copilot wrote:

In this step, `build_date` is written to `$GITHUB_ENV`, but the subsequent log line references `${BUILD_DATE}` (uppercase) which is never set in the shell, so the value will always be empty/misleading. Set a shell variable (e.g., `BUILD_DATE=$(date ...)`) before writing to `$GITHUB_ENV`, or echo `${{ env.build_date }}` in a later step.
```suggestion
          BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
          echo "build_date=${BUILD_DATE}" >> $GITHUB_ENV
```

alfonder reviewed (changes requested):
Please, update your PR and I'll be happy to merge it.

azinchen committed:
Address PR review: fix BUILD_DATE in deploy.yml, revert nginx formatting

- deploy.yml: assign BUILD_DATE to a shell variable so the log line
  echoing ${BUILD_DATE} prints the actual value
- rootfs/etc/nginx/nginx.conf: revert formatting-only changes
- rootfs/etc/nginx/conf.d/default.conf: revert formatting-only changes
